cmd/podman/artifact/rm.go

```diff
@@ -11,40 +11,62 @@ import (
 
 var (
 	rmCmd = &cobra.Command{
-		Use:               "rm ARTIFACT",
-		Short:             "Remove an OCI artifact",
-		Long:              "Remove an OCI from local storage",
-		RunE:              rm,
-		Aliases:           []string{"remove"},
-		Args:              cobra.ExactArgs(1),
+		Use:     "rm [options] ARTIFACT",
+		Short:   "Remove an OCI artifact",
+		Long:    "Remove an OCI from local storage",
+		RunE:    rm,
+		Aliases: []string{"remove"},
+		Args: func(cmd *cobra.Command, args []string) error { //nolint: gocritic
+			return checkAllAndArgs(cmd, args)
+		},
 		ValidArgsFunction: common.AutocompleteArtifacts,
-		Example:           `podman artifact rm quay.io/myimage/myartifact:latest`,
-		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Example: `podman artifact rm quay.io/myimage/myartifact:latest
+podman artifact rm -a`,
+		Annotations: map[string]string{registry.EngineMode: registry.ABIMode},
 	}
-	// The lint avoid here is because someday soon we will need flags for
-	// this command
-	rmFlag = rmFlagType{} //nolint:unused
+
+	rmOptions = entities.ArtifactRemoveOptions{}
 )
 
-// TODO at some point force will be a required option; but this cannot be
-// until we have artifacts being consumed by other parts of libpod like
-// volumes
-type rmFlagType struct { //nolint:unused
-	force bool
+func rmFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVarP(&rmOptions.All, "all", "a", false, "Remove all artifacts")
 }
-
 func init() {
 	registry.Commands = append(registry.Commands, registry.CliCommand{
 		Command: rmCmd,
 		Parent:  artifactCmd,
 	})
+	rmFlags(rmCmd)
 }
 
 func rm(cmd *cobra.Command, args []string) error {
-	artifactRemoveReport, err := registry.ImageEngine().ArtifactRm(registry.Context(), args[0], entities.ArtifactRemoveOptions{})
+	var nameOrID string
+	if len(args) > 0 {
+		nameOrID = args[0]
+	}
+	artifactRemoveReport, err := registry.ImageEngine().ArtifactRm(registry.Context(), nameOrID, rmOptions)
 	if err != nil {
 		return err
 	}
-	fmt.Println(artifactRemoveReport.ArtfactDigest.Encoded())
+	for _, d := range artifactRemoveReport.ArtfactDigests {
+		fmt.Printf("Deleted: %s\n", d.Encoded())
```

Comment:
Why is this changed? I don't see the value of printing Deleted on every line. Of course the stuff we print has been deleted, I see that image rm has that but that prints two different things Untagged and Deleted.
If we only ever print the digest I don't see the point of the prefix.

In general I don't belive this should be part of this change at all, it has nothing do with --all, so if you actually want this then it should be in its own separate commit.

Reply:
I'm happy to revert but I was modelling after images.

Reply:
Yes but it has nothing to do with this change so it should not be part of this IMO. If you like that output, fine for me but "sneaking" it in while adding --all is not correct.